lib/vmdb/settings/database_source.rb

```diff
@@ -3,23 +3,51 @@ class Settings
     class DatabaseSource
       include Vmdb::Logging
 
-      attr_reader :resource
+      attr_reader :resource, :settings_change_holder
 
-      def initialize(resource)
-        @resource = resource
+      def initialize(resource_to_get_settings_about, class_to_call_settings_changes_on)
+        @resource = resource_to_get_settings_about
+        @settings_change_holder = class_to_call_settings_changes_on
+      end
+
+      def self.sources_for(resource_to_get_settings_about)
+        return [] if resource_to_get_settings_about.nil?
+        hierarchy_index = SETTINGS_HIERARCHY.index(resource_to_get_settings_about.class.name.to_sym)
+        SETTINGS_HIERARCHY[0..hierarchy_index].collect do |class_to_call_settings_changes_on|
```

Comment:
And this can be `klass` or something similar.  Even `settings_changes_holder`, though I'm not a fan of that name (but I can't come up with anything better yet)